src/dyn_model.jl

```diff
@@ -0,0 +1,307 @@
+# Code for building and running a power management problem
+
+
+# ===
+# POWER MANAGEMENT PROBLEM
+# ===
+
+TAU = 1e-5
+INIT_COND = 0
+
+mutable struct PowerManagementProblem
+    problem::Problem
+    p
+    g
+    s
+    params
+end
+
+"""
+    PowerManagementProblem(f, d, pmax, gmax, A, ds; τ=1e-5)
+
+Set up a power management problem with generation costs `f`, demand `d`,
+maximum power flow `pmax`, maximum generation `gmax`, and network
+incidence matrix `A`. `ds` is the change in battery state of charge. It should
+be set to 0 if no battery is in the network. 
+
+The parameter `τ` is a regularization weight used to make the problem
+strongly convex by adding τ ∑ᵢ pᵢ² to the objective. Parameters fq and fl 
+are the Quadratic and Linear terms in the generator costs. P is the maximum battery charge and 
+discharge power (i.e. the maximum amount of energy that can be exchanged with a nodal battery
+during a given timestep).
+
+TODO: check that we're not messing up the ordering!
+"""
+function PowerManagementProblem(fq, fl, d, pmax, gmax, A, ds; τ=TAU)
+    n, m = size(A)
+    g = Variable(n)
+    p = Variable(m)
+
+    # add variable-specific constraints
+    add_constraints(p, [
+        -p <= pmax, 
+        p <= pmax, 
+    ])
+    add_constraints!(g, [
```

Comment:
I think `add_constraints!` only works for `Convex.Problem` objects. We might need to use `add_constraint!` and add constraints one at a time for `Convex.Variable` objects. Need to double check.